streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -750,7 +750,7 @@ protected void maybeClean() {
         long now = time.milliseconds();
 
         if (now > lastCleanMs + cleanTimeMs) {
```

Comment:
Yes, but the same check is applied within `stateDirectory.cleanRemovedTasks(cleanTimeMs);` -- it's redundant, isn't it?